clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -1188,6 +1203,14 @@ boolean isEmpty() {
                     inFlightAcknowledgements.isEmpty();
         }
 
+        /**
+         * Resets the timer with the new deadline and resets the RequestState.
```

Comment:
This comment says "new deadline" but I don't see the setting of a NEW deadline in here. Please clarify.